dart/dynamics/Shape.h

```diff
@@ -191,29 +142,19 @@ class Shape : public virtual common::Subject
   /// Instruct this shape to update its data
   virtual void refreshData();
 
-  /// \brief
-  virtual void draw(renderer::RenderInterface* _ri = nullptr,
-                    const Eigen::Vector4d& _color = Eigen::Vector4d::Ones(),
-                    bool _useDefaultColor = true) const = 0;
-
-  /// Pass in true to prevent this shape from being rendered. Pass in false to
-  /// allow it to render again.
-  void setHidden(bool _hide=true);
+  /// Notify that the color (rgba) of this shape has updated
+  virtual void notifyColorUpdate(const Eigen::Vector4d& color);
```

Comment:
I think materials should just be key/value stores that map string keys to [vec4 | texture | string] values. You can try to force everything into an overengineered canonical set of material attributes, but nobody will be happy. There are other vertex attributes you might want to have besides colors, as well. The tricky bit is that for efficiency a renderer should keep the vertex data GPU resident and only update it when it changes, which means that the viewer needs to be notified of mesh vertex data changes. Likewise with textures.